resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/tpr/sparkJobResourceController.scala

```diff
@@ -0,0 +1,316 @@
+    extends Logging with sparkJobResourceController {
+  private val kubeMaster = k8sClient.getMasterUrl.toString
+  private val httpClient: OkHttpClient =
+    extractHttpClientFromK8sClient(k8sClient.asInstanceOf[BaseClient])
+  private val namespace = k8sClient.getNamespace
+
+  private implicit val formats: Formats = DefaultFormats + JobStateSerDe
+  private implicit val ec: ThreadPoolExecutor = ThreadUtils
+    .newDaemonCachedThreadPool("tpr-watcher-pool")
+
+  /**
+   * This method creates a resource of kind "SparkJob"
+   * @return Unit
+   * @throws IOException Due to failure from execute call
+   * @throws SparkException when POST request is unsuccessful
+   */
+  override def createJobObject(name: String, status: Status): Unit = {
+    val resourceObject =
+      SparkJobState(s"$TPR_API_GROUP/$TPR_API_VERSION", TPR_KIND, Metadata(name), status)
+    val payload = parse(write(resourceObject))
+    val requestBody = RequestBody
+      .create(MediaType.parse("application/json"), compact(render(payload)))
+    val requestPathSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs")
+    val url = generateHttpUrl(requestPathSegments)
+    val request = new Request.Builder()
+      .post(requestBody)
+      .url(url)
+      .build()
+
+    logDebug(s"Create SparkJobResource Request: $request")
+    var response: Response = null
+    try {
+      response = httpClient.newCall(request).execute()
+    } catch {
+      case x: IOException =>
+        val msg =
+          s"Failed to post resource $name. ${x.getMessage}. ${compact(render(payload))}"
+        logError(msg)
+        response.close()
+        throw new SparkException(msg)
+    }
+    completeRequestWithExceptionIfNotSuccessful(
+      "post",
+      response,
+      Option(Seq(name, response.toString, compact(render(payload)))))
+    logDebug(s"Successfully posted resource $name: " +
+      s"${pretty(render(parse(write(resourceObject))))}")
+    response.body().close()
+  }
+
+  /**
+   * This method deletes a resource of kind "SparkJob" with the specified name
+   * @return Unit
+   * @throws IOException Due to failure from execute call
+   * @throws SparkException when DELETE request is unsuccessful
+   */
+  override def deleteJobObject(tprObjectName: String): Unit = {
+    val requestPathSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", tprObjectName)
+    val url = generateHttpUrl(requestPathSegments)
+    val request = new Request.Builder()
+      .delete()
+      .url(url)
+      .build()
+
+    logDebug(s"Delete Request: $request")
+    var response: Response = null
+    try {
+      response = httpClient.newCall(request).execute()
+    } catch {
+      case x: IOException =>
+        val msg =
+          s"Failed to delete resource. ${x.getMessage}."
+        logError(msg)
+        response.close()
```

Comment:
Actually since `ResponseBody` implements closeable, look into how we can use `Utils.tryWithResource(...) {...}`.